src-plugins/medBinaryOperation/medBinaryOperatorBase.cpp

```diff
@@ -0,0 +1,261 @@
+        }
+        else
+        {
+            res = medAbstractProcess::PIXEL_TYPE;
+        }
+    }
+    return res;
+}
+
+template <class ImageType, class ImageType2> int medBinaryOperatorBase::runProcess()
+{
+    if ( !m_inputA->data() || !m_inputB->data())
+    {
+        return DTK_FAILURE;
+    }
+    QApplication::setOverrideCursor(Qt::WaitCursor);
+    QApplication::processEvents();
+
+    typedef itk::Image<unsigned char, 3> ImageTypeOutput;
+    ImageTypeOutput::Pointer imageA;
+    ImageTypeOutput::Pointer imageB;
+
+    if (m_inputA->identifier() != "itkDataImageUChar3")
+    {
+        typename ImageType::Pointer imA = dynamic_cast< ImageType*>((itk::Object*)(m_inputA->data()));
+        typedef itk::CastImageFilter< ImageType, ImageTypeOutput > CasterType;
+        typename CasterType::Pointer caster = CasterType::New();
+        caster->SetInput(imA);
+        caster->Update();
+        imageA = caster->GetOutput();
+    }
+    else
+    {
+        imageA = dynamic_cast< ImageTypeOutput*>((itk::Object*)(m_inputA->data()));
+    }
+
+    if (m_inputB->identifier() != "itkDataImageUChar3")
+    {
+        typename ImageType2::Pointer imB = dynamic_cast< ImageType2*>((itk::Object*)(m_inputB->data()));
+        typedef itk::CastImageFilter< ImageType2, ImageTypeOutput > CasterType;
+        typename CasterType::Pointer caster = CasterType::New();
+        caster->SetInput(imB);
+        caster->Update();
+        imageB = caster->GetOutput();
+    }
+    else
+    {
+        imageB = dynamic_cast< ImageTypeOutput*>((itk::Object*)(m_inputB->data()));
+    }
+
+    if (!imageA || !imageB)
+    {
+        qDebug()<<"Problems appear during the cast to ITK image type";
+        return DTK_FAILURE;
+    }
+    
+    typedef itk::InPlaceImageFilter< ImageTypeOutput, ImageTypeOutput >  FilterType;
+    typename FilterType::Pointer filter;
+    
+    if(description() == "AND")
+    {
+        typedef itk::AndImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> AndImageFilterType;
+        filter = AndImageFilterType::New();
+    }
+
+    else if(description() == "OR")
+    {
+        typedef itk::OrImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> OrImageFilterType;
+        filter = OrImageFilterType::New();
+    }
+
+    else if(description() == "XOR")
+    {
+        typedef itk::XorImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> XorImageFilterType;
+        filter = XorImageFilterType::New();
+    }
+
+    else
+    {
+        qDebug()<<"Wrong binary operator";
+        return DTK_FAILURE;
+    }
+
+    filter->SetInput(0, imageA);
+    filter->SetInput(1, imageB);
+
+    try
+    {
+        filter->Update();
+    }
+    catch( itk::ExceptionObject & err )
+    {
+        std::cerr << "ExceptionObject caught in "<< metaObject()->className() << std::endl;
+        std::cerr << err << std::endl;
+        QApplication::restoreOverrideCursor();
+        QApplication::processEvents();
```

Comment:
Actually, I tried the cursors but my processes were too fast to see anything, so I just considered that since they were there before ...
But I can remove them, if needed.